src/c-writer.cc

```diff
@@ -539,6 +453,12 @@ std::string CWriter::LegalizeName(string_view name) {
   for (size_t i = 1; i < name.size(); ++i)
     result += isalnum(name[i]) ? name[i] : '_';
 
+  // In addition to containing valid characters for C, we must also avoid
+  // colliding with things C cares about, such as reserved words (e.g. "void")
+  // or a function name like main() (which a compiler will  complain about if we
+  // define it with another type). To avoid such problems, prefix.
+  result = "__w2c_" + result;
```

Comment:
Although it's unlikely to conflict with anything, this is technically a reserved identifier (anything with `__`).  Maybe just `w2c_` as the prefix?

Reply:
Oh heh, I was hoping by using reserved ids we'd be less likely to conflict with user code. But yeah, there's a risk with system code. `w2c_` sgtm, but it is short enough that I imagine someone will eventually collide with it...

Reply:
Should be OK though, since the names here only need to avoid colliding with system headers. If  someone actually uses the name `foo` and `w2c_foo` in the same program, it should be fine since the first will be mangled to `w2c_foo` and the second to `w2c_foo_1`, I believe.

Reply:
Oh I see, yes, that sounds safe then.